bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/TeaserImpl.java

```diff
@@ -44,10 +47,34 @@
     @ScriptVariable
     protected Page currentPage;
 
+    /**
+     * The current component.
+     */
+    @ScriptVariable
+    private Component component;
+
+    @PostConstruct
+    protected void initModel() {
+
+        super.initModel();
+
+        if ((super.getTitle()!=null && !super.getTitle().isEmpty()) || super.isActionsEnabled()) {
+            super.hiddenImageResourceProperties.add(Link.PN_LINK_URL);
+        }
+
+        if (hasImage()) {
+            super.setImageResource(component, request.getResource(), super.hiddenImageResourceProperties, null);
+        }
+    }
+
     @Override
+    @JsonInclude(JsonInclude.Include.NON_NULL)
     @Nullable
     public Link getLink() {
-        return link.orElse(null);
+        if (!super.isActionsEnabled())
```

Comment:
I haven't tested it but I think that this will break following requirement:
The teaser title, description and image can be inherited from the linked page, or from the page linked in the first call to action.
So the link should point to the first CTA if CTA are enabled.
I think we should remove following requirement as both requirements exclude each other:
In the JSON, remove the linkURL when CTA are active.
@gabrielwalt WDYT?

Reply:
Addressed on this commit: https://github.com/adobe/aem-core-wcm-components/commit/512dd3672e80db4a7ec593b3e60093bcb4969dad